site/public/ts/utils/tools/SelectionTool.ts

```diff
@@ -238,6 +253,12 @@ export class SelectionTool extends Tool {
         return this.p2.copy();
     }
 
+    set onSelectionChanged(func: {(): void}) {
+        this.callbacks.push(func);
+    }
+    private selectionChanged() {
```

Comment:
selectionsChanged()*
Also let's keep private helper methods underneath the constructor my usual layout for classes are:

[variables]
[constructor]
[private methods]
[protected methods]
[public modifiers]
[public setters]
[public getters]
[abstract methods]